features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/RuntimeConfiguration.jsx

```diff
@@ -298,11 +331,13 @@ export default function RuntimeConfiguration() {
     const { api, updateAPI } = useContext(APIContext);
     const isAsyncAPI = api.type === 'WS' || api.type === 'WEBSUB' || api.type === 'SSE';
     const isWebSub = api.type === 'WEBSUB';
-    api.websubSubscriptionConfiguration = api.websubSubscriptionConfiguration || {
-        secret: '',
-        signingAlgorithm: '',
-        signatureHeader: '',
-    };
+    if (!api.isAPIProduct()) {
+        api.websubSubscriptionConfiguration = api.websubSubscriptionConfiguration || {
```

Comment:
I think this `api` object is coming from the APIContext , so altering it directly is not valid

Reply:
removed